webpack.config.js

```diff
@@ -340,6 +341,80 @@ const settingsPage = {
 	],
 };
 
+const styles = {
+	...sharedConfig,
+	entry: {
+		'admin-tables': './assets/css/src/admin-tables.css',
+		'amp-block-editor': './assets/css/src/amp-block-editor.css',
+		'amp-customizer': './assets/css/src/amp-customizer.css',
+		'amp-customizer-legacy': './assets/css/src/amp-customizer-legacy.css',
+		'amp-default': './assets/css/src/amp-default.css',
+		'amp-icons': './assets/css/src/amp-icons.css',
+		'amp-mobile-version-switcher': './assets/css/src/amp-mobile-version-switcher.css',
+		'amp-playlist-shortcode': './assets/css/src/amp-playlist-shortcode.css',
+		'amp-post-meta-box': './assets/css/src/amp-post-meta-box.css',
+		'amp-validation-error-taxonomy': './assets/css/src/amp-validation-error-taxonomy.css',
+		'amp-validation-single-error-url': './assets/css/src/amp-validation-single-error-url.css',
+		'amp-validation-tooltips': './assets/css/src/amp-validation-tooltips.css',
+	},
+	module: {
+		...sharedConfig.module,
+		rules: sharedConfig.module.rules.map(
+			( rule ) => {
+				const cssLoader = Array.isArray( rule.use ) && rule.use.find(
+					( loader ) => loader.loader && loader.loader.includes( '/css-loader' ),
+				);
+
+				/**
+				 * Prevent "Module not found: Error: Can't resolve ..."
+				 * being thrown for `url()` CSS rules.
+				 */
+				if ( cssLoader ) {
+					cssLoader.options = {
+						...cssLoader?.options,
+						url: false,
+					};
+				}
+
+				return rule;
+			},
+		),
+	},
+	plugins: [
+		...sharedConfig.plugins.filter(
+			( plugin ) => plugin.constructor.name !== 'DependencyExtractionWebpackPlugin',
+		),
+		new WebpackBar( {
+			name: 'Styles',
+			color: '#b763ff',
+		} ),
+	],
+};
+
+/**
+ * Get JS chunk names for a list of style entry points.
+ *
+ * @param {Array} entries List of style entry points.
+ *
+ * @return {Array} List of JS chunk names.
+ */
+const extractChunkNamesForStylesheets = ( entries ) =>
+	entries
+		.map( ( entryPath ) => {
+			const match = entryPath.match( /^.*\/(.*)\..*$/ );
+
+			return match && match[ 1 ] ? `${ match[ 1 ] }.js` : null;
+		} )
+		.filter( Boolean );
```

Comment:
Does the `match` regex need to be this generic? Since all the entry points are CSS files this could be made simpler:

```suggestion
			const match = entryPath.match( /([^\/]+)\.css$/ );
			return `${ match[ 1 ] }.js`;
		} );
```

Reply:
The entry points may also be SCSS files, but I guess the regex could still get simplified to something like:

```regex
/([^\/]+)\.s?css$/
```

Reply:
Ah right, that works as well.

Reply:
@pierlon Would you apply that change and verify it produces the expected output? Then we can merge this PR if I'm not mistaken, at least for beta2.

Reply:
I've noticed that an empty JS chunk is also being generated for the [`wp-components` entrypoint](https://github.com/ampproject/amp-wp/blob/e64576a56f386283cdcf41799c25b23ae2ed22fe/webpack.config.js#L293) on the "Settings page" webpack config. We can either move it to the "Styles" webpack config, or extract these set of changes into a new webpack plugin so that we can apply it to multiple webpack configs.

Reply:
I think we can go for the former right now, and then deal with the latter later on. I can create an issue for that if needed.